src/middlewared/middlewared/pytest/unit/plugins/test_datastore.py

```diff
@@ -636,6 +639,44 @@ class DefaultModel(Model):
         assert (await ds.query("test.default", [], {"get": True}))["string"] == "VALUE"
 
 
+class StringPrimaryKeyModel(Model):
+    __tablename__ = 'test_stringprimarykey'
+
+    string_id = sa.Column(sa.String(100), primary_key=True)
+    value = sa.Column(sa.Integer(), nullable=True)
+
+
+class IntegerPrimaryKeyModel(Model):
+    __tablename__ = 'test_integerprimarykey'
+
+    integer_id = sa.Column(sa.String(100), primary_key=True)
+    value = sa.Column(sa.Integer(), nullable=True)
+
+
+@pytest.mark.asyncio
+async def test__insert_string_pk_record():
+    async with datastore_test() as ds:
+        pk = await ds.insert("test.stringprimarykey", {"string_id": "unique_key", "value": 1})
+
+        assert len(await ds.query("test.stringprimarykey", [["string_id", "=", pk]])) == 1
```

Comment:
Won't it be better to assert returned value from ds.insert?

Reply:
I think this is better as we ensure that the value returned is actually the primary key as reflected by the database and not somehow just being returned from `ds.insert`

Reply:
We can do both asserts

Reply:
Tests like this tend to hide situations where two wrongs make one right (but are still wrongs)